components/hab/src/command/pkg/export/docker.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ffi::OsString;
+
+use common::ui::UI;
+
+use error::Result;
+
+const EXPORT_CMD: &'static str = "hab-pkg-export-docker";
+
+pub fn start(ui: &mut UI, args: Vec<OsString>) -> Result<()> {
+    inner::start(ui, args)
+}
+
+#[cfg(target_os = "linux")]
+mod inner {
+    use std::ffi::OsString;
+    use std::path::PathBuf;
+    use std::str::FromStr;
+
+    use common::ui::UI;
+    use hcore::crypto::{init, default_cache_key_path};
+    use hcore::env as henv;
+    use hcore::fs::find_command;
+    use hcore::os::process;
+    use hcore::package::PackageIdent;
+
+    use error::{Error, Result};
+    use exec;
+    use VERSION;
+    use super::EXPORT_CMD;
+
+    const EXPORT_CMD_ENVVAR: &'static str = "HAB_PKG_EXPORT_DOCKER_BINARY";
+    const EXPORT_PKG_IDENT: &'static str = "core/hab-pkg-export-docker";
+    const EXPORT_PKG_IDENT_ENVVAR: &'static str = "HAB_PKG_EXPORT_DOCKER_PKG_IDENT";
+
+    pub fn start(ui: &mut UI, args: Vec<OsString>) -> Result<()> {
+        let command = match henv::var(EXPORT_CMD_ENVVAR) {
+            Ok(command) => PathBuf::from(command),
+            Err(_) => {
+                init();
```

Comment:
Should we be starting this earlier in the function? I think I made changes recently so this wouldn't panic if you called it more than once - can't remember exactly

Reply:
We could do a future refactor which just jams that early in `hab`. We need this for the install logic/verify code and have been only calling it when necessary. But yeah, I've never been 100% happy with that. It's possible that the overhead of that call is so small that it isn't an issue.